lib/text/simple_text_displayer.js

```diff
@@ -22,66 +26,246 @@ goog.require('shaka.text.Utils');
  */
 shaka.text.SimpleTextDisplayer = class {
   /**
-   * @param {HTMLMediaElement} video
-   * @param {string} label
+   * @param {shaka.Player} player
    */
-  constructor(video, label) {
-    /** @private {HTMLMediaElement} */
-    this.video_ = video;
+  constructor(player) {
+    /** @private {?shaka.Player} */
+    this.player_ = player;
 
-    /** @private {string} */
-    this.textTrackLabel_ = label;
+    /** @private {?HTMLMediaElement} */
+    this.video_ = null;
 
-    /** @private {TextTrack} */
-    this.textTrack_ = null;
+    /** @private {!Map<number, !HTMLTrackElement>} */
+    this.trackNodes_ = new Map();
+
+    /** @private {number} */
+    this.trackId_ = -1;
+
+    /** @private {boolean} */
+    this.visible_ = false;
+
+    /** @private {boolean} */
+    this.hasMicroTask_ = false;
+
+    /** @private */
+    this.onLoaded_ = () => {
+      this.player_.removeEventListener(
+          shaka.util.FakeEvent.EventName.Loaded,
+          this.onLoaded_,
+      );
+      if (
+        this.player_.getLoadMode() === 2 // shaka.Player.LoadMode.MEDIA_SOURCE
+      ) {
+        this.video_ = this.player_.getMediaElement();
+        this.player_.addEventListener(
+            shaka.util.FakeEvent.EventName.Unloading,
+            this.onUnloading_,
+        );
+        this.player_.addEventListener(
+            shaka.util.FakeEvent.EventName.TextChanged,
+            this.onTextChanged_,
+        );
+        this.video_.textTracks.addEventListener('change', this.onChange_);
+        this.onTextChanged_();
+      }
+    };
 
-    // TODO: Test that in all cases, the built-in CC controls in the video
-    // element are toggling our TextTrack.
+    /** @private */
+    this.onUnloading_ = () => {
+      this.player_.removeEventListener(
+          shaka.util.FakeEvent.EventName.Unloading,
+          this.onUnloading_,
+      );
```

Comment:
```suggestion
      this.player_.removeEventListener(
          shaka.util.FakeEvent.EventName.Unloading,
          this.onUnloading_);
```
We prefer this format, please update all code to this format.